src/server/main_service.cc

```diff
@@ -1646,6 +1648,10 @@ void Service::DispatchMC(const MemcacheParser::Command& cmd, std::string_view va
     if (cmd.type == MemcacheParser::GETS) {
       dfly_cntx->conn_state.memcache_flag |= ConnectionState::FETCH_CAS_VER;
     }
+
+    if (cmd.type == MemcacheParser::GAT) {
+      dfly_cntx->conn_state.memcache_flag |= ConnectionState::SET_EXPIRY | expire_ts << 2;
+    }
```

Comment:
Another option instead of doing this would to to store the expiry using `EXAT ...` in the last two args as done for `SET` instead of stashing it in the state. 

A problem with that approach is that since the command is passed directly to `MGETS`, the last two args are processed as part of whatever shards they are assigned to, and the code becomes complicated because it has to skip searching for those two keys.